lib/ocpp/common/websocket/websocket_tls.cpp

```diff
@@ -302,15 +304,34 @@ tls_context WebsocketTLS::on_tls_init(std::string hostname, websocketpp::connect
         }
 
         context->set_verify_mode(boost::asio::ssl::verify_peer);
+
         if (this->connection_options.verify_csms_common_name) {
```

Comment:
Issues with the libwebsocket impl, the hostname verification is skipped, looking into the issue.

Reply:
The issue was too long to look into, an workaround has been added, now the check is done manually since it seems that libwebsockets either skips certificate CN verification or it over-writes the verification parameter. 'X509_check_host' is manually used until a proper fix will be found.

However, with the manual verification in place, a very clear error message can be print.